pkg/kv/iterators.go

```diff
@@ -253,24 +253,29 @@ type PartitionIterator struct {
 }
 
 func NewPartitionIterator(ctx context.Context, store Store, msgType protoreflect.MessageType, partitionKey string) (*PartitionIterator, error) {
-	itr, err := store.Scan(ctx, []byte(partitionKey), []byte(""))
-	if err != nil {
-		return nil, err
-	}
 	return &PartitionIterator{
 		ctx:          ctx,
 		store:        store,
 		msgType:      msgType,
-		itr:          itr,
 		partitionKey: partitionKey,
 	}, nil
 }
 
+func getNewIter(ctx context.Context, store Store, partitionKey string, start []byte) (EntriesIterator, error) {
+	return store.Scan(ctx, []byte(partitionKey), start)
+}
+
 func (p *PartitionIterator) Next() bool {
 	if p.Err() != nil {
 		return false
 	}
 	p.value = nil
+	if p.itr == nil {
+		p.itr, p.err = getNewIter(p.ctx, p.store, p.partitionKey, []byte(""))
+		if p.Err() != nil {
+			return false
+		}
+	}
```

Comment:
In the case 'p.itr' is nil it means we failed to initialize or seek to a key - we should render an error or return false.